src/main/java/org/ow2/proactive/catalog/service/BucketGrantService.java

```diff
@@ -0,0 +1,409 @@
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.catalog.service;
+
+import static org.ow2.proactive.catalog.util.AccessType.*;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import org.ow2.proactive.catalog.dto.BucketGrantMetadata;
+import org.ow2.proactive.catalog.dto.BucketMetadata;
+import org.ow2.proactive.catalog.repository.BucketGrantRepository;
+import org.ow2.proactive.catalog.repository.BucketRepository;
+import org.ow2.proactive.catalog.repository.CatalogObjectGrantRepository;
+import org.ow2.proactive.catalog.repository.entity.BucketEntity;
+import org.ow2.proactive.catalog.repository.entity.BucketGrantEntity;
+import org.ow2.proactive.catalog.repository.entity.CatalogObjectGrantEntity;
+import org.ow2.proactive.catalog.service.exception.BucketGrantAlreadyExistsException;
+import org.ow2.proactive.catalog.service.model.AuthenticatedUser;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import lombok.extern.log4j.Log4j2;
+
+
+/**
+ * @author ActiveEon Team
+ */
+@Log4j2
+@Service
+@Transactional
+public class BucketGrantService {
+
+    @Autowired
+    private BucketRepository bucketRepository;
+
+    @Autowired
+    private BucketGrantRepository bucketGrantRepository;
+
+    @Autowired
+    private CatalogObjectGrantRepository catalogObjectGrantRepository;
+
+    @Autowired
+    private CatalogObjectGrantService catalogObjectGrantService;
+
+    @Autowired
+    private GrantAccessTypeHelperService grantAccessTypeHelperService;
+
+    /**
+     *
+     * Get the grants assigned to a user and his group
+     *
+     * @param username username
+     * @param userGroup user group
+     * @return all grants that are assigned to a specific username and his group
+     */
+    public List<BucketGrantMetadata> getAllAssignedGrantsForUserAndHisGroup(String username, String userGroup) {
```

Comment:
consider having a list or set of groups in parameter and usage of the SQL expression 'in'